### ISSUE TITLE
Subpages have incomplete tx_templavoilaplus_flex and break moving content around

### ISSUE THREAD

**Issue by jpmschuler** [T3Voila/templavoilaplus#418]
TL;DR: v8.1.1 needs a `tx_templavoilaplus_flex` which reflects the whole DataStructure in order to work correctly, however upgraded legacy systems won't provide this for subpages not having a TOBJ/MAP set. 

-----

I have a page template with multiple content areas, lets call them top, main, side, bottom. For most pages `main` and `side` are filled. This results in a `pages.tx_templavoilaplus_flex` like
```
<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<T3FlexForms>
    <data>
        <sheet index="sDEF">
            <language index="lDEF">
                <field index="field_main">
                    <value index="vDEF">86816</value>
                </field>
                <field index="field_side">
                    <value index="vDEF">86815</value>
                </field>
            </language>
        </sheet>
    </data>
</T3FlexForms>
```
With this I can move around content elements between these two columns, e.g. moving the main item to `side`, which results in
```
<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<T3FlexForms>
    <data>
        <sheet index="sDEF">
            <language index="lDEF">
                <field index="field_main">
                    <value index="vDEF"></value>
                </field>
                <field index="field_side">
                    <value index="vDEF">86816,86815</value>
                </field>
            </language>
        </sheet>
    </data>
</T3FlexForms>
```
However I can't move content elements to `top` or `bottom` (results in moved item flashing "red" instead of "green").

If I edit the page record (e.g. in list module) and save it without changing anything `pages.tx_templavoilaplus_flex` field gets populated with the empty areas added from the datastructure:
```
<?xml version="1.0" encoding="utf-8" standalone="yes" ?>
<T3FlexForms>
    <data>
        <sheet index="sDEF">
            <language index="lDEF">
                <field index="field_main">
                    <value index="vDEF"></value>
                </field>
                <field index="field_side">
                    <value index="vDEF">86816,86815</value>
                </field>
                <field index="field_top">
                    <value index="vDEF"></value>
                </field>
                <field index="field_bottom">
                    <value index="vDEF"></value>
                </field>
            </language>
        </sheet>
    </data>
</T3FlexForms>
```
Afterwards all moving around works.

This seems not to be the case for pages which had `tx_templavoila_tobj` set during upgrade (thus now have `tx_templavoilaplus_map` set) as they were "touched" during the upgrade process, however for subpages just inheriting the mapping no such touch took place and they have this kind of "trimmed" flex field breaking the moving around.

This probably needs to be fixed in the code as else this meant adding fields to an existing template/mapping won't work.

A quick hot fix of course is just to edit and save each page - however that might be automated.
